ways-of-working/rfc-244-use-automated-accessibility-checks-in-feature-tests.md

```diff
@@ -0,0 +1,88 @@
+# Use automated accessibility checks in feature tests
+
+## Summary
+
+Accessibility is core to our mission, but it's easy to let it slip in moments of
+pressure, or to treat it as something to fix just before going live. Tools exist
+that can help us integrate accessibility checks into our feature tests without
+relying on memory or time, alerting us as soon as something goes wrong. We
+should use these tools where possible to make building accessible services
+easier.
+
+## Problem
+
+One of dxw's core principles is to [make everything we do
+accessible][dxw-accessibility-principle], and the first point of [our mission
+statement][dxw-mission-statement] is to:
+
+> build services that fit seamlessly into user’s lives, and that make public
+> services usable and _accessible to all_ [emphasis added] – especially those
+> most in need
+
+We take accessibility seriously in our client projects: designers, developers,
+and members of other disciplines within a sprinting team strive to consider
+accessibility within every iteration, and we conduct accessibility audits as
+standard.
+
+However, in the pressure of an busy sprint, especially when deadlines are
+looming, it's easy to forget to check for accessibility issues. We tend also to
+hold off on more holistic accessibility audits until nearing the end of a
+statement of work, or when a service is about to go live.
+
+We should work by the adage that accessibility should never be an afterthought.
+
+## Proposal
+
+We SHOULD integrate automated accessibility checks into feature tests using
+axe-core where possible. axe is one of the most popular accessibility testing
+suites available, with widespread usage and good language support. We have a
+good level of familiarity with the axe suite within dxw, and axe DevTools, which
+runs on axe-core, is recommended in our own internal accessibility training.
+axe, including axe-core, is open source. While axe has premium products (we
+subscribe to axe DevTools Pro), axe-core is free to use.
+
+We MAY use an alternative automated accessibility testing tool in environments
+where axe is unavailable or another tool is better suited.
+
+We MUST continue to perform manual checks as part of a holistic approach to
```

Comment:
What to do for manual checks has felt unclear to me in the past and some direction would be good if we are going to have this as a MUST. I agree we should do it, but how?

For me it's quite hard to work with a service everyday and be able to spot the nuances during a review of how the service presents itself: does the text of our links match the page title that you land on? Are the table scopes set correctly in a way that describes their content to screen readers? 

The big test has been a formal accessibility audit which gives us great feedback but can take weeks to schedule and get the feedback from. We have to be strategic about when we schedule this as we haven't the budget to do it on a regular basis. In the past we've reserved this until going into public beta. 

I've had success in the past by inviting fresh eyes into the team to review parts of the service. They spotted things we'd missed and were easier to schedule than a full test. We could schedule sooner and more frequently.

Do we need to set out some options? Do we need to tailor our response depending on the phase of the project?

Reply:
Yeah - perhaps this could be added as a list after the current point to clarify what the minimum expectation is (without taking overall focus away from the automated testing)?

> We MUST continue to perform manual checks as part of a holistic approach to accessibility testing. The [WCAG AA standard](https://www.gov.uk/service-manual/helping-people-to-use-your-service/understanding-wcag#meeting-government-accessibility-requirements) is [the minimum requirement](https://www.gov.uk/service-manual/helping-people-to-use-your-service/making-your-service-accessible-an-introduction#meeting-government-accessibility-requirements) for the types of services we work on. Automated accessibility checks don't cover 100% of the AA requirements - [axe-core claims around 57% coverage](https://www.deque.com/automated-accessibility-testing-coverage) - therefore manual checks are still necessary to plug the gaps. At minimum:
> - we MUST do a full accessibility audit before going to public beta
> - we MAY invite people to join a sprinting team for an accessibility review before public beta
> - we SHOULD be trying to spot accessibility issues during code review

Reply:
Might also be worth flagging in the summary that there are some brief recommendations on our approach to manual testing

Reply:
I guess it would be good to have some guidelines of what makes good manual (and achievable) testing during development - so that the audit is less of a hurdle. 

This is obviously a bigger task, and something we might want to iterate on - but hopefully the content in the accessibility manual work might help towards this.

Reply:
I've added a fixup commit to address this: https://github.com/dxw/tech-team-rfcs/pull/244/commits/2a6242a08a05d792fb46362c7266d8c47d3fdef7

Will leave the conversation unresolved for now in case there are other thoughts